src/volttron/server/aip.py

```diff
@@ -483,56 +484,74 @@ def land_agent(self, agent_wheel):
             raise
         return agent_uuid
 
+    def backup_agent_data(self, agent_uuid, publickey, secretkey, vip_identity):
+        backup_agent_file = None
+        if agent_uuid:
+            _log.debug(f"There is an existing agent {agent_uuid}")
+            old_agent_data_dir = self.get_agent_data_dir(agent_uuid)
+            if os.listdir(old_agent_data_dir):
+                # And there is data to backup
+                backup_agent_file = "/tmp/{}.tar.gz".format(agent_uuid)
+                with tarfile.open(backup_agent_file, "w:gz") as tar:
+                    tar.add(old_agent_data_dir, arcname=os.path.sep)  # os.path.basename(source_dir))
+            # current agent's keystore overrides any keystore data passed
+            keystore = self.__get_agent_keystore__(vip_identity, publickey, secretkey)
+            publickey = keystore.public
+            secretkey = keystore.secret
+
+            _log.info('Removing previous version of agent "{}"\n'.format(vip_identity))
+            self.remove_agent(agent_uuid)
+        return backup_agent_file, publickey, secretkey
+
+    @staticmethod
+    def restore_agent_data_from_tgz(source_file, output_dir):
```

Comment:
Should this helper method be in some utils module?

Reply:
ad of now this is used only by aip